include/xtensor/xstrides.hpp

```diff
@@ -461,10 +461,10 @@ namespace xt
         return detail::unravel_noexcept(index, strides, l);
     }
 
-    template <class S>
-    inline get_strides_t<S> ravel_from_strides(const S& index, const S& strides)
+    template <class S, class T>
+    inline get_value_type_t<T> ravel_from_strides(const T& index, const S& strides)
```

Comment:
Good catch for the return type.
Why allowing different types for the index and the strides? Is that for having mixed arithmetic or for having different containers for the strides an index?

Reply:
Signedness, since `get_strides_type` ha a member array whose value_type is `ptrdiff_t`: https://github.com/xtensor-stack/xtensor/blob/5b6f0996a2429d3b0d1bcdd101f22e65322090d9/include/xtensor/xutils.hpp#L830.
I guess it is wished for to be able to pass a `xt::uvector` or a `xt::uvector` as an index...

Reply:
Ha indeed.